### ISSUE TITLE
Changing mode at runtime allows invalid combinations

### ISSUE THREAD

**Issue by bohlender** [aquynh/capstone#1214]
As expected, using the combined mode `CS_MODE_32 | CS_MODE_BIG_ENDIAN`, fails to initialize Capstone 3.0 (i.e. `cs_open` returns an invalid handle). However, when setting the very same combined mode at runtime, there is no indication of error. Is this intentional? Seems more like a bug to me, or am I missing something?

The following example illustrates the issue:
```C
#include <stdio.h>
#include <inttypes.h>
#include <capstone/capstone.h>
#include <assert.h>

#define CODE "\x55\x48\x8b\x05\xb8\x13\x00\x00"

int main(void) {
    csh handle;
    cs_insn *insn;
    size_t count;

    if (cs_open(CS_ARCH_X86, CS_MODE_32, &handle) != CS_ERR_OK) // Valid mode
        return -1;
    cs_err err = cs_option(handle, CS_OPT_MODE, CS_MODE_32 | CS_MODE_BIG_ENDIAN); // Invalid mode
    assert(err == CS_ERR_OK);
    count = cs_disasm(handle, CODE, sizeof(CODE)-1, 0x1000, 0, &insn);
    if (count > 0) {
        printf("Ok\n");
        cs_free(insn, count);
    } else
        printf("Error\n");
    cs_close(&handle);
}
```
I expected it to fail with the invalid mode set.

**Comment by aquynh** [comment]
Is this the issue with latest code from master branch? If not, please
specify the version.

**Comment by bohlender** [comment]
Thanks for the fast reply. I was using the version currently available in the ubuntu repositories (3.0.4):
```
$ apt-cache policy libcapstone3
libcapstone3:
  Installed: 3.0.4-0.2
  Candidate: 3.0.4-0.2
  Version table:
 *** 3.0.4-0.2 500
        500 http://de.archive.ubuntu.com/ubuntu xenial/universe amd64 Packages
        100 /var/lib/dpkg/status
```
However, I've tried your suggestion of using the current master version and it turns out that the issue is indeed resolved there. In fact, it turns out that the issue was _resolved in 3.0.5_ (while still existent in 3.0.5-rc3). I'll just stick with the stable 3.0.5.

**Comment by aquynh** [comment]
Yes, hopefully distros update to the latest version soon.
